packages/transactional-adapters/transactional-adapter-typeorm/src/lib/transactional-adapter-typeorm.ts

```diff
@@ -0,0 +1,38 @@
+import { DataSource, EntityManager } from 'typeorm';
+import {
+    TypeOrmTransactionOptions,
+    TypeOrmTransactionalAdapterOptions,
+} from './transactional-options';
+import { TransactionalAdapter } from '@nestjs-cls/transactional';
+
+export class TransactionalAdapterTypeOrm
+    implements
+        TransactionalAdapter<
+            DataSource,
+            DataSource | EntityManager,
```

Comment:
It would be probably better if the return type of `tx` was the same. Can we make it `EntityManager` in each case?

On line `36`, you could use `getFallbackInstance: () => typeORMInstance.manager`

Also please rename `typeORMInstance` to `dataSource`, so it's not confusing.

Reply:
Ok